# cellrank-MCP

Natural language interface for scRNA-Seq analysis with cellrank through MCP.

## ðŸª© What can it do?

- IO module like read and write scRNA-Seq data
- Preprocessing module,like filtering, quality control, normalization, scaling, highly-variable genes, PCA, Neighbors,...
- Tool module, like clustering, differential expression etc.
- Plotting module, like violin, heatmap, dotplot

## â“ Who is this for?

- Anyone who wants to do scRNA-Seq analysis natural language!
- Agent developers who want to call cellrank's functions for their applications

## ðŸŒ Where to use it?

You can use cellrank-mcp in most AI clients, plugins, or agent frameworks that support the MCP:

- AI clients, like Cherry Studio
- Plugins, like Cline
- Agent frameworks, like Agno 

## ðŸŽ¬ Demo

A demo showing scRNA-Seq cell cluster analysis in a AI client Cherry Studio using natural language based on cellrank-mcp



## ðŸŽï¸ Quickstart

### Install

Install from PyPI
```
pip install cellrank-mcp
```
you can test it by running
```
cellrank-mcp run
```

#### run scnapy-server locally
Refer to the following configuration in your MCP client:

```
"mcpServers": {
  "cellrank-mcp": {
    "command": "cellrank-mcp",
    "args": [
      "run"
    ]
  }
}
```

#### run scnapy-server remotely
Refer to the following configuration in your MCP client:

run it in your server
```
cellrank-mcp run --transport shttp --port 8000
```

Then configure your MCP client, like this:
```
http://localhost:8000/mcp
```

## ðŸ¤ Contributing

If you have any questions, welcome to submit an issue, or contact me(hsh-me@outlook.com). Contributions to the code are also welcome!

## Citing

If you use cellRank-mcp in for your research, please consider citing  following work: 
> Weiler, P., Lange, M., Klein, M. et al. CellRank 2: unified fate mapping in multiview single-cell data. Nat Methods 21, 1196â€“1205 (2024). https://doi.org/10.1038/s41592-024-02303-9
